src/NotebookViewer/NotebookViewer.tsx

```diff
@@ -10,35 +10,42 @@ import {
 import { IGalleryItem, JunoClient } from "../Juno/JunoClient";
 import * as GalleryUtils from "../Utils/GalleryUtils";
 import { GalleryHeaderComponent } from "../Explorer/Controls/Header/GalleryHeaderComponent";
+import { FileSystemUtil } from "../Explorer/Notebook/FileSystemUtil";
 
 const onInit = async () => {
   initializeIcons();
   await initializeConfiguration();
   const galleryViewerProps = GalleryUtils.getGalleryViewerProps(window.location.search);
   const notebookViewerProps = GalleryUtils.getNotebookViewerProps(window.location.search);
   const backNavigationText = galleryViewerProps.selectedTab && GalleryUtils.getTabTitle(galleryViewerProps.selectedTab);
+  const hideInputs = notebookViewerProps.hideInputs;
 
   const notebookUrl = decodeURIComponent(notebookViewerProps.notebookUrl);
-  render(notebookUrl, backNavigationText);
+  render(notebookUrl, backNavigationText, hideInputs);
 
   const galleryItemId = notebookViewerProps.galleryItemId;
   if (galleryItemId) {
     const junoClient = new JunoClient();
     const notebook = await junoClient.getNotebook(galleryItemId);
-    render(notebookUrl, backNavigationText, notebook.data);
+    render(notebookUrl, backNavigationText, hideInputs, notebook.data);
```

Comment:
nit: this could be a 2nd call to `render()`? Isn't it better to put the first `render()` inside an `else` of this case (iow if `galleryItemId` is `undefined`)?

Reply:
Yes this would be a 2nd call to render. Since React anyway takes care of diff-ing the DOM I thought it's better to call render as soon as we can. In our case first render can be done immediately after the page loads, but the second render needs to wait for `junoClient.getNotebook` to finish. This makes the loading experience more snappy.

Reply:
Good point. I agree!